airflow/providers/amazon/aws/auth_manager/views/auth.py

```diff
@@ -0,0 +1,132 @@
+    def login(self):
+        """Start logging process."""
+        saml_auth = self._init_saml_auth()
+        return redirect(saml_auth.login())
+
+    @expose("/logout")
+    def logout(self):
+        """Start logout process."""
+        session.clear()
+        saml_auth = self._init_saml_auth()
+
+        return redirect(saml_auth.logout())
+
+    @csrf.exempt
+    @expose("/login_callback", methods=("GET", "POST"))
+    def login_callback(self):
+        """Callback where the user is redirected to after successful login."""
+        saml_auth = self._init_saml_auth()
+        saml_auth.process_response()
+        errors = saml_auth.get_errors()
+        is_authenticated = saml_auth.is_authenticated()
+        if not is_authenticated:
+            error_reason = saml_auth.get_last_error_reason()
+            logging.error("Failed to authenticate")
+            logging.error("Errors: %s", errors)
+            logging.error("Error reason: %s", error_reason)
+            raise AirflowException(f"Failed to authenticate: {error_reason}")
+
+        attributes = saml_auth.get_attributes()
+        user = AwsAuthManagerUser(
+            user_id=attributes["id"][0],
+            groups=attributes["groups"],
+            username=saml_auth.get_nameid(),
+            email=attributes["email"][0],
+        )
+        session["aws_user"] = user
+
+        return redirect(url_for("Airflow.index"))
+
+    @expose("/login_metadata")
+    def login_metadata(self):
+        saml_auth = self._init_saml_auth()
+        settings = saml_auth.get_settings()
+        metadata = settings.get_sp_metadata()
+        errors = settings.validate_metadata(metadata)
+
+        if len(errors) == 0:
+            resp = make_response(metadata, 200)
+            resp.headers["Content-Type"] = "text/xml"
+        else:
+            resp = make_response(", ".join(errors), 500)
+        return resp
+
+    @staticmethod
+    def _prepare_flask_request() -> dict:
+        return {
+            "https": "on" if request.scheme == "https" else "off",
+            "http_host": request.host,
+            "script_name": request.path,
+            "get_data": request.args.copy(),
+            "post_data": request.form.copy(),
+        }
+
+    def _init_saml_auth(self) -> OneLogin_Saml2_Auth:
+        request_data = self._prepare_flask_request()
+        base_url = conf.get(section="webserver", key="base_url")
+        settings = {
+            # We want to keep this flag on in case of errors.
+            # It provides an error reasons, if turned off, it does not
+            "debug": True,
+            "sp": {
+                "entityId": f"{base_url}/login_metadata",
+                "assertionConsumerService": {
+                    "url": f"{base_url}/login_callback",
+                    "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+                },
+                "singleLogoutService": {
+                    "url": f"{base_url}/logout_callback",
```

Comment:
I don't see login_callback defined above, where do we get that from?

Reply:
`login_callback` is defined as route in this file (line 64). However, your comment make me thing we might not need `logout_callback`. Identity center does not support single logout. Which means, when users click on "log out", they get redirected to identity center where they can log out, but the actual log out does not happen. I talked to Identity center guys and it is intended. Anyway, I dont think the logout feature should not be very used with a workforce population of users

Reply:
ugh, my bad, I meant logout_callback in my original message. I could see login_callback defined but not logout.

> they get redirected to identity center where they can log out, but the actual log out does not happen

What does that mean it does not happen?

Reply:
They do not get logged out. So basically that would work that way:
- User is signed in Airflow using AWS auth manager
- User clicks on "log out"
- User's session in Airflow is cleared, hence the user is logged out in Airflow (but the session still exist in Identity Center)
- User is redirected to Identity Center (AWS portal landing page as they call it) where they are not logged out but have the option to do so (through a link)

Reply:
This is an AWS Service limitation, right?  Not something we are likely to be able to influence or implement?